### PR TITLE
Converted initSkipLink.js to TypeScript & added unit tests

### PR DESCRIPTION
**Pull request by Julietadeboye**
Fix for [Issue 9541](https://github.com/wagtail/wagtail/issues/9541)

Converted the js file to typescript.
Added JSDOC
Added a test file.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: client/src/includes/initSkipLink.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Application } from '@hotwired/stimulus';
+
```
Comment (lb-) [inline client/src/includes/initSkipLink.test.js:2]: This test needs work - the Stimulus application import will not work (I know this was pulled from future proposed code but please revisit). 

Also the test itself should test what happens when you click the button. Not just the initial state where there are no focused elements.

File: client/src/includes/initSkipLink.ts [hunk 2]
```diff
@@ -1,17 +1,25 @@
 const initSkipLink = () => {
-  // Inspired by https://github.com/selfthinker/dokuwiki_template_writr/blob/master/js/skip-link-focus-fix.js
+  /**
+   * Appears at the top left corner of the admin page with the tab button is clicked.
+   * Used to provide an accessible skip button for keyboard control so that users can easily navigate to the main content without having to navigate a long list of navigation links.
+   *
+   * Inspired by https://github.com/selfthinker/dokuwiki_template_writr/blob/master/js/skip-link-focus-fix.js
+   *
+   */
```
Comment (lb-) [inline client/src/includes/initSkipLink.ts:9]: This JSDOC needs to be above the function declaration to be used correctly by code editors.

**Review by lb- — CHANGES_REQUESTED** [review]
@Julietadeboye a great start but needs a bit more work - mainly for the unit tests. 

Thanks for taking the time to get this moving.

File: client/src/includes/initSkipLink.test.js (hunk 1 above)
Comment (Julietadeboye) [reply client/src/includes/initSkipLink.test.js:2]: Probably the reason the test i wrote for the button kept failing, i will redo thanks

File: client/src/includes/initSkipLink.ts (hunk 2 above)
Comment (Julietadeboye) [reply client/src/includes/initSkipLink.ts:9]: Noted

File: client/src/includes/initSkipLink.ts [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+
```
Comment (lb-) [inline client/src/includes/initSkipLink.ts:8]: This line break should be removed, not a huge issue though, I will do this when merging.

```suggestion
```

File: client/src/includes/initSkipLink.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import initSkipLink from './initSkipLink';
+
+describe('initSkipLink', () => {
+  document.body.innerHTML = `
+  <div>
+  <a id="test" class="skiplink button" href="#main" data-skiplink="">Skip to main content</a>
+  </div>
+  <main id="main" >
```
Comment (lb-) [inline client/src/includes/initSkipLink.test.js:8]: Should be ok for the test but just a reminder to keep an eye out for malformed DOM when writing tests like this.
```suggestion
  <main id="main">
```

File: client/src/includes/initSkipLink.test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import initSkipLink from './initSkipLink';
```
Comment (lb-) [inline client/src/includes/initSkipLink.test.js:1]: We should use named exports - unfortunately this is not documented anywhere yet - I will fix this up.

**Review by lb- — APPROVED** [review]
Thanks @Julietadeboye  - this is looking good - I know that you have had some issues with prettier so I will fix this up in my local and merge this in.
I have a few small clean up items commented but can fix those up.

The core functionality and unit tests are working great, thanks so much for taking the time to contribute this improvement.

**Comment by Julietadeboye** [comment]
Thanks LB for the review!

**Pull request merged**